File: dataextract.py
import pymysql
import json

# Read the JSON data
def read_json(file_path):
    with open(file_path, 'r') as file:
        data = json.load(file)
    return data

# Function to connect to the database
def connect_to_db():
    connection = pymysql.connect(
        host='127.0.0.1',  # Use the IP address instead of 'localhost'
        user='root',        # Your MySQL username
        password='',  # Your MySQL password
        database='biodiversity', # Your database name
        port=3306           # Default MySQL port
    )
    return connection

# Function to insert data into the database
def insert_data(connection, json_data):
    try:
        with connection.cursor() as cursor:
            for item in json_data:
                # Check if 'indicator' and 'country' fields exist and are dictionaries
                if isinstance(item, dict) and 'indicator' in item and isinstance(item['indicator'], dict) and 'value' in item['indicator'] \
                        and 'country' in item and isinstance(item['country'], dict) and 'value' in item['country']:
                    
                    indicator_value = item['indicator']['value']
                    country_value = item['country']['value']
                    date_value = item.get('date', None)  # Default to None if 'date' is missing
                    value_value = item.get('value', None)  # Default to None if 'value' is missing
                    
                    # SQL query to insert data into the table (adjust the table and columns as needed)
                    sql = """
                        INSERT INTO marine_data (indicator, country, date, value)
                        VALUES (%s, %s, %s, %s)
                    """
                    
                    # Execute the query
                    cursor.execute(sql, (indicator_value, country_value, date_value, value_value))
                else:
                    print(f"Missing or malformed 'indicator' or 'country' in record: {item}")
            
            # Commit the transaction
            connection.commit()
            print(f"Inserted {len(json_data)} records successfully.")
    
    except Exception as e:
        print(f"Error: {e}")
        connection.rollback()

# Main function to load JSON and insert into DB
def main():
    # Path to your JSON file
    file_path = 'data.json'  # Replace with the correct path to your JSON file
    
    # Read JSON data
    json_data = read_json(file_path)
    
    # Connect to the database
    connection = connect_to_db()
    
    # Insert data into the database
    insert_data(connection, json_data)
    
    # Close the connection
    connection.close()

if __name__ == "__main__":
    main()

File: dataextract.java
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.File;
import java.io.IOException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class dataextract {

    public static void main(String[] args) {
        // Path to the JSON file
        String jsonFilePath = "data.json"; // Replace with your actual file path

        // Database credentials
        String url = "jdbc:mysql://localhost:3306/biobiversity"; // Replace with your DB URL
        String user = "root"; // Replace with your DB username
        String password = ""; // Replace with your DB password
        
        // Read JSON and insert data into the database
        try {
            // Parse JSON file
            JsonNode rootNode = readJsonFile(jsonFilePath);

            // Establish database connection
            try (Connection conn = DriverManager.getConnection(url, user, password)) {
                // Loop through the JSON array and insert each record
                if (rootNode.isArray()) {
                    for (JsonNode recordNode : rootNode) {
                        String indicatorValue = recordNode.get("indicator").get("value").asText();
                        String countryValue = recordNode.get("country").get("value").asText();
                        String date = recordNode.get("date").asText();
                        Double value = recordNode.get("value").isNull() ? null : recordNode.get("value").asDouble();

                        // Insert data into the database
                        insertDataIntoDatabase(conn, indicatorValue, countryValue, date, value);
                    }
                }
            } catch (SQLException e) {
                e.printStackTrace();
            }

        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    // Method to read JSON file
    private static JsonNode readJsonFile(String filePath) throws IOException {
        ObjectMapper objectMapper = new ObjectMapper();
        return objectMapper.readTree(new File(filePath));
    }

    // Method to insert data into the database
    private static void insertDataIntoDatabase(Connection conn, String indicator, String country, String date, Double value) {
        String insertSQL = "INSERT INTO indicators_data (indicator, country, date, value) VALUES (?, ?, ?, ?)";

        try (PreparedStatement stmt = conn.prepareStatement(insertSQL)) {
            stmt.setString(1, indicator);
            stmt.setString(2, country);
            stmt.setString(3, date);
            if (value != null) {
                stmt.setDouble(4, value);
            } else {
                stmt.setNull(4, java.sql.Types.DOUBLE);
            }
            stmt.executeUpdate();
            System.out.println("Inserted: " + indicator + ", " + country + ", " + date + ", " + value);
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}
